src/pages/vaults/VaultPositionsTable.tsx

```diff
@@ -19,98 +20,111 @@ import { TableCell } from '@/components/Table/TableCell';
+                </div>
+              </TableCell>
+            );
+          },
         },
         {
           columnKey: 'size',
-          getCellValue: (row) => row.currentPosition.usdc,
+          getCellValue: (row) => row.currentPosition?.usdc,
           label: stringGetter({ key: STRING_KEYS.SIZE }),
-          renderCell: ({ currentPosition, marketId, asset }) => (
+          renderCell: ({ currentPosition, marketId }) => (
             <TableCell stacked>
-              <Output value={currentPosition.usdc} type={OutputType.Fiat} fractionDigits={0} />
+              <Output value={currentPosition?.usdc} type={OutputType.Fiat} fractionDigits={0} />
               <Output
-                value={currentPosition.asset}
+                value={currentPosition?.asset}
                 type={OutputType.Asset}
-                tag={asset.id}
-                fractionDigits={marketsData[marketId]?.configs?.stepSizeDecimals}
+                tag={marketsData[marketId ?? '']?.assetId}
+                fractionDigits={marketsData[marketId ?? '']?.configs?.stepSizeDecimals}
               />
             </TableCell>
           ),
         },
         {
           columnKey: 'pnl',
-          getCellValue: (row) => row.thirtyDayPnl.absolute,
+          getCellValue: (row) => row.thirtyDayPnl?.absolute,
           label: stringGetter({ key: STRING_KEYS.VAULT_THIRTY_DAY_PNL }),
-          renderCell: ({ thirtyDayPnl }) => (
-            <TableCell
-              stacked
-              slotRight={
-                <div style={{ width: 50, height: 50 }} tw="ml-0.5">
-                  <SparklineChart
-                    data={thirtyDayPnl.sparklinePoints.map((elem, index) => ({
-                      x: index + 1,
-                      y: elem,
-                    }))}
-                    xAccessor={(datum) => datum.x}
-                    yAccessor={(datum) => datum.y}
-                    positive={thirtyDayPnl.absolute > 0}
-                  />
-                </div>
-              }
-            >
-              <$OutputSigned
-                sign={getNumberSign(thirtyDayPnl.absolute)}
-                value={thirtyDayPnl.absolute}
-                type={OutputType.Fiat}
-                fractionDigits={0}
-              />
-              <Output
-                value={thirtyDayPnl.percent}
-                type={OutputType.Percent}
-                showSign={ShowSign.Both}
-              />
-            </TableCell>
-          ),
+          renderCell: ({ thirtyDayPnl: thirtyDayPnlRaw }) => {
+            const thirtyDayPnl = thirtyDayPnlRaw ?? {
+              absolute: 0,
+              percent: 0,
+              sparklinePoints: undefined,
+            };
+            return (
+              <TableCell
+                stacked
+                slotRight={
+                  <div style={{ width: 50, height: 50 }} tw="ml-0.5">
+                    {thirtyDayPnl.sparklinePoints != null && (
```

Comment:
from the default on line `90` it looks like `thirtyDayPnl.sparklinePoints` could be `undefined`. would this crash on line `102` : `thirtyDayPnl.sparklinePoints.toArray().map`

Reply:
It's guarded by the check on this line I believe.

Reply:
It checks for null but not undefined. I might be missing something 😅

Reply:
!= null catches both iirc, !== null would only be null

Reply:
TIL 😳